internal/util/kubernetes/unstructured_util.go

```diff
@@ -1,13 +1,309 @@
 package kubernetes
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
+	"strings"
 
+	"github.com/numaproj/numaplane/internal/util"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/numaproj/numaplane/internal/util/logger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
 )
 
 // this file contains utility functions for working with Unstructured types
```

Comment:
Not sure what you think. I had trouble figuring out what should go in `dynamic_client.go` vs `unstructured_util.go` so I just put everything in here and removed `dynamic_client.go`

Reply:
This is just a pure remove? anything new added?

Reply:
I added List functions: `ListCR()` and `ListUnstructuredCR()`, and now there's an "unstructured" version of each of the functions: Get, List, Create, Update

Reply:
I also added `CompareSpecs()` and `DeepCopy()` functions